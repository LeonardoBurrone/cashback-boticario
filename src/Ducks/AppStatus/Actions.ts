// TODO: testar actions
import { AppStatusActionTypes, ChangeIsLoggedInAction, ChangeLoadingAction, ChangeLoadingMessageAction } from './types';

export const changeIsLoggedInAction = (payload: boolean): ChangeIsLoggedInAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_IS_LOGGED_IN
});

export const changeLoadingAction = (payload: boolean): ChangeLoadingAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_LOADING
});

export const changeLoadingMessageAction = (payload: string): ChangeLoadingMessageAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_LOADING_MESSAGE
});
