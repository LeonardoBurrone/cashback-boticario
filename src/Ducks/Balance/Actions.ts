import { BalanceActionTypes, ChangeBalanceAction, ChangeBalanceErrorAction, FetchBalanceAction } from './types';

export const changeBalanceAction = (payload: string): ChangeBalanceAction => ({
  payload,
  type: BalanceActionTypes.CHANGE_BALANCE
});

export const changeBalanceErrorAction = (payload: boolean): ChangeBalanceErrorAction => ({
  payload,
  type: BalanceActionTypes.CHANGE_BALANCE_ERROR
});

export const fetchBalanceAction = (): FetchBalanceAction => ({
  type: BalanceActionTypes.FETCH_BALANCE
});
