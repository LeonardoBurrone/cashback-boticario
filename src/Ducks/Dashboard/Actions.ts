// TODO: testar actions
import { ChangePuchasesAction, FetchPuschasedItemsAction, DashboardActionTypes, Purchase } from './types';

export const changePuchasesAction = (payload: Purchase[]): ChangePuchasesAction => ({
  payload,
  type: DashboardActionTypes.CHANGE_PURCHASES
});

export const fetchPurchasedItemsAction = (): FetchPuschasedItemsAction => ({
  type: DashboardActionTypes.FETCH_PURCHASED_ITEMS
});
