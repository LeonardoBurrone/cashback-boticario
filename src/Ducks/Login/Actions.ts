import { AuthenticateAction, LoginAction, LoginActionTypes, LogoutAction } from './types';

export const authenticateAction = (): AuthenticateAction => ({
  type: LoginActionTypes.AUTHENTICATE
});

export const loginAction = (payload: { email: string; password: string }): LoginAction => ({
  payload,
  type: LoginActionTypes.LOGIN
});

export const logoutAction = (): LogoutAction => ({
  type: LoginActionTypes.LOGOUT
});
