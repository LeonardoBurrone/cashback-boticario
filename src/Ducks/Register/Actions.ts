import { RegisterActionTypes, RegisterPurchaseAction } from './types';

export const registerPurchaseAction = (payload: {
  code: string;
  date: string;
  price: string;
}): RegisterPurchaseAction => ({
  payload,
  type: RegisterActionTypes.REGISTER_PURCHASE
});
