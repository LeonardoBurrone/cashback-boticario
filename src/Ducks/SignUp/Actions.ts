// TODO: testar actions
import { SignActionTypes, SignUpAction } from './types';

export const signUpAction = (payload: {
  document: string;
  email: string;
  name: string;
  password: string;
}): SignUpAction => ({
  payload,
  type: SignActionTypes.SIGN_UP
});
