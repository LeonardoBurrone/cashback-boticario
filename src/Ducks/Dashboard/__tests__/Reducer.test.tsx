import { changePuchasesAction, fetchPurchasesErrorAction } from '../Actions';
import dashboardReducer, { initialState } from '../Reducer';

describe('Test Dashboard Reducer', () => {
  it('Check the initial state', () => {
    expect(dashboardReducer(initialState, {})).toMatchSnapshot();
  });

  it('Check reducer state after "changePuchasesAction" was dispatched', () => {
    expect(
      dashboardReducer(
        initialState,
        changePuchasesAction([
          {
            cashbackPercentage: 0.01,
            cashbackValue: 1,
            date: '2021-04-01T08:15:30-03:00',
            id: 123,
            price: 100,
            status: 'Approved'
          },
          {
            cashbackPercentage: 0.01,
            cashbackValue: 1,
            date: '2021-04-01T08:11:30-03:00',
            id: 123,
            price: 50,
            status: 'Validating'
          },
          {
            cashbackPercentage: 0.01,
            cashbackValue: 1,
            date: '2021-04-01T08:19:30-03:00',
            id: 123,
            price: 30,
            status: 'Rejected'
          }
        ])
      )
    ).toMatchSnapshot();
  });

  it('Check reducer state after "fetchPurchasesErrorAction" was dispatched', () => {
    expect(dashboardReducer(initialState, fetchPurchasesErrorAction(true))).toMatchSnapshot();
  });
});
