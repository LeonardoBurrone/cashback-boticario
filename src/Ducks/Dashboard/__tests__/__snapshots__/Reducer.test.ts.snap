// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard Reducer Check reducer state after "changePuchasesAction" was dispatched 1`] = `
Object {
  "fetchPurchasesError": false,
  "purchases": Array [
    Object {
      "cashbackPercentage": 0.01,
      "cashbackValue": 1,
      "date": "2021-04-01T08:15:30-03:00",
      "id": 123,
      "price": 100,
      "status": "Approved",
    },
    Object {
      "cashbackPercentage": 0.01,
      "cashbackValue": 1,
      "date": "2021-04-01T08:11:30-03:00",
      "id": 123,
      "price": 50,
      "status": "Validating",
    },
    Object {
      "cashbackPercentage": 0.01,
      "cashbackValue": 1,
      "date": "2021-04-01T08:19:30-03:00",
      "id": 123,
      "price": 30,
      "status": "Rejected",
    },
  ],
}
`;

exports[`Test Dashboard Reducer Check reducer state after "fetchPurchasesErrorAction" was dispatched 1`] = `
Object {
  "fetchPurchasesError": true,
  "purchases": Array [],
}
`;

exports[`Test Dashboard Reducer Check the initial state 1`] = `
Object {
  "fetchPurchasesError": false,
  "purchases": Array [],
}
`;
